xwayland/selection/selection.c

```diff
@@ -32,16 +45,59 @@ void xwm_selection_transfer_destroy_property_reply(
 	transfer->property_reply = NULL;
 }
 
-void xwm_selection_transfer_init(struct wlr_xwm_selection_transfer *transfer) {
+void xwm_selection_transfer_init(struct wlr_xwm_selection_transfer *transfer,
```

Comment:
Idea for a future patch: make this perform the allocation and rename to `xwm_selection_transfer_create`.